### PR TITLE
Fix FileHelpers.ResolveRelativePath

### PR DESCRIPTION
**Pull request by cra0zy**
Fixes: #5581

### CODE REVIEW

File: MonoGame.Framework/Utilities/FileHelpers.cs [hunk 1]
```diff
@@ -44,7 +47,7 @@ public static string ResolveRelativePath(string filePath, string relativeFile)
                 filePath = ForwardSlashString + filePath;
 
             // Get a uri for filePath using the file:// schema and no host.
-            var src = new Uri("file://" + filePath);
+            var src = new Uri("file://" + UrlEncode(filePath));
 
             var dst = new Uri(src, relativeFile);
```
Comment (KonajuGames) [inline MonoGame.Framework/Utilities/FileHelpers.cs]: I would suggest encoding the relative path as well.  It could contain characters in need of encoding.

Comment (cra0zy) [inline MonoGame.Framework/Utilities/FileHelpers.cs]: That would create an incorrect file path.

Comment (cra0zy) [inline MonoGame.Framework/Utilities/FileHelpers.cs]: Oh no, wait, I got confused, you are right.

File: MonoGame.Framework/Utilities/FileHelpers.cs [hunk 2]
```diff
@@ -3,12 +3,15 @@
...
-
+using System.IO;
+using System.Text;
+
 namespace Microsoft.Xna.Framework.Utilities
 {
     internal static class FileHelpers
-    {
+    {
+        private const string UrlSafeChars = "._-;/?:@&=+&,";
```
Comment (mrhelmut) [inline MonoGame.Framework/Utilities/FileHelpers.cs]: ```&``` isn't safe. :)

Comment (cra0zy) [inline MonoGame.Framework/Utilities/FileHelpers.cs]: You are right, I included the query URL components which I shouldn't have, fixed now.

**Comment by mrhelmut** [comment]
It works perfectly, thank you @cra0zy ! :)

**Comment by dellis1972** [comment]
looks ok to me @tomspilman @KonajuGames

**Comment by KonajuGames** [comment]
It generates more garbage than I would ideally like, and the magic numbers could be changed to their char equivalents, but functionally it does work.

**Comment by cra0zy** [comment]
I'll see what I can do.

**Comment by cra0zy** [comment]
@KonajuGames Better?

**Comment by KonajuGames** [comment]
That does improve it a bit.  Once the tests have passed, I'm happy to merge it.

**Comment by KonajuGames** [comment]
Merging. Thanks @cra0zy

**Pull request merged**